### PR TITLE
Add Entrenching Component

### PR DESCRIPTION
**Pull request by Mike-MF**
- Adds entrenching tool abilities to backpacks with shovels.

(Requires GRAD update to work.)

### CODE REVIEW

**Review by jonpas — APPROVED** [review]
It's just a config entry, can merge if GRAD's API is finalized.

**Comment by veteran29** [comment]
No need to wait, it's ACE feature which GRAD just does not "see" right now.

File: addons/entrenching/CfgVehicles.hpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    class CUP_O_RUS_Patrol_bag_BeigeDigital_Shovel: Bag_Base {
+        ace_trenches_entrenchingTool = 1;
+    };
+    class CUP_O_RUS_Patrol_bag_Desert_Shovel: Bag_Base {
+        ace_trenches_entrenchingTool = 1;
+    };
+    class CUP_O_RUS_Patrol_bag_Summer_Shovel: Bag_Base {
+        ace_trenches_entrenchingTool = 1;
+    };
+    class CUP_O_RUS_Patrol_bag_Desert_Shovel: Bag_Base {
```
Comment (veteran29) [inline addons/entrenching/CfgVehicles.hpp:12]: This is already defined above.

Comment (Mike-MF) [reply addons/entrenching/CfgVehicles.hpp:12]: I fixed it.

**Pull request merged**